src/backends/mod.rs

```diff
@@ -1,38 +1,81 @@
 use super::types::{Message, Partition, Position, Topic};
 use std::collections::{HashMap, HashSet};
+use thiserror::Error;
 
 pub mod kafka;
 pub mod local;
 pub mod storages;
 
-#[derive(Debug, Clone)]
-pub struct ConsumerClosed;
+#[derive(Error, Debug)]
+pub enum SubscriptionError {
+    #[error("The consumer is closed")]
+    ConsumerClosed,
 
-#[derive(Debug, Clone)]
-pub struct EndOfPartition;
+    #[error("Subscription error")]
+    BrokerError { source: Box<dyn std::error::Error> },
```

Comment:
This is the part I am not sure about. 
This is a trait, it has multiple implementations, the actual errors raised by the individual implementation are implementation specific (`KafkaError` for the kafka implementation and implementation specific errors for the local one). 
So the errors that come from the implementations can only be known here as `std::error::Error`. I feel there should be an easier way to do this, though at least this works and does not expose the implementation specific details.